db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2162,9 +2158,11 @@ Status DBImpl::AtomicFlushMemTables(
     }
     WaitForPendingWrites();
 
+    ColumnFamilyData* cfd_flush_disabled = nullptr;
     if (immutable_db_options_.replication_log_listener) {
-      // If replication_log_listener is installed the only thing we are allowed
-      // to do is flush all column families.
+      cfd_flush_disabled = GetAnyCFWithFlushDisabled();
+      // If replication_log_listener is installed the only thing we are
+      // allowed to do is flush all column families.
```

Comment:
Let's actually allow the manual flush even if `disable_flush` is `true`. In fact, I would call it `disable_auto_flush` and have it exactly correspond to `disable_auto_compaction`, which disables automatic compactions, but does allow manual compaction.

It makes the code much simpler and is makes the option more useful.